cli/cmd/network_create.go

```diff
@@ -14,9 +24,91 @@ func (r *runners) InitNetworkCreate(parent *cobra.Command) *cobra.Command {
 	}
 	parent.AddCommand(cmd)
 
+	cmd.Flags().StringVar(&r.args.createNetworkName, "name", "", "Network name (defaults to random name)")
+	cmd.Flags().StringVar(&r.args.createNetworkTTL, "ttl", "", "Network TTL (duration, max 48h)")
+	cmd.Flags().DurationVar(&r.args.createNetworkWaitDuration, "wait", time.Second*0, "Wait duration for Network to be ready (leave empty to not wait)")
+
+	cmd.Flags().BoolVar(&r.args.createNetworkDryRun, "dry-run", false, "Dry run")
+
+	cmd.Flags().StringVar(&r.outputFormat, "output", "table", "The output format to use. One of: json|table|wide (default: table)")
+
 	return cmd
 }
 
 func (r *runners) createNetwork(_ *cobra.Command, args []string) error {
-	return nil
+	if r.args.createNetworkName == "" {
+		r.args.createNetworkName = namesgenerator.GetRandomName(0)
+	}
+
+	opts := kotsclient.CreateNetworkOpts{
+		Name:   r.args.createNetworkName,
+		TTL:    r.args.createNetworkTTL,
+		DryRun: r.args.createNetworkDryRun,
+	}
+
+	network, err := r.createAndWaitForNetwork(opts)
+	if err != nil {
+		if errors.Cause(err) == ErrVMWaitDurationExceeded {
+			defer func() {
```

Comment:
Why is a `defer` needed?

Also, `os.Exit` doesn't return anything, so you don't need to wrap it in `func()` you can just do a one liner:

```
defer os.Exit(124)
```